packages/shared_preferences/lib/shared_preferences.dart

```diff
@@ -38,6 +37,26 @@ class SharedPreferences {
     return _completer.future;
   }
 
+  static SharedPreferencesPlatform _platform;
+
+  /// Sets a custom [SharedPreferencesPlatform].
+  ///
+  /// This property can be set to use a custom platform implementation.
+  static set platform(SharedPreferencesPlatform platform) {
+    assert(_platform == null);
+    _platform = platform;
+  }
+
+  /// The SharedPreferences platform that's used by the plugin.
+  ///
+  /// The default value is [MethodChannelSharedPreferences] on Android or iOS.
+  static SharedPreferencesPlatform get platform {
```

Comment:
I think that we should keep this static field in `SharedPreferencesPlatform` so that platform implementations can depend solely on the platform interface package.